File: 4기 코딩테스트_220615/4기 - 2nd 코딩테스트/CountEqualWithDividedNumber/CountEqualWithDividedNumber_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.medium.CountEqualWithDividedNumber.CountEqualWithDividedNumber import \
    solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        nums = [4, 1, 2, 1, 4]
        d = 2

        # when
        result = solution(nums, d)

        # then
        self.assertEqual(1, result)

    def test_2(self):
        # given
        nums = [6, 5, 4, 3, 2, 1, 2, 3, 4, 5, 6]
        d = 3

        # when
        result = solution(nums, d)

        # then
        self.assertEqual(2, result)

    def test_3(self):
        # given
        nums = [2]
        d = 2

        # when
        result = solution(nums, d)

        # then
        self.assertEqual(0, result)

    def test_4(self):
        # given
        nums = []
        d = 3

        # when
        result = solution(nums, d)

        # then
        self.assertEqual(0, result)

    def test_5(self):
        # given
        nums = [10, 10]
        d = 10

        # when
        result = solution(nums, d)

        # then
        self.assertEqual(1, result)

    def test_6(self):
        # given
        nums = [10, 8, 6, 4, 2, 10, 8, 6, 4, 2]
        d = 2

        # when
        result = solution(nums, d)

        # then
        self.assertEqual(5, result)


if __name__ == '__main__':
    unittest.main()

File: 4기 코딩테스트_220615/4기 - 2nd 코딩테스트/CountEqualWithDividedNumber/CountEqualWithDividedNumber_test.js
import assert from 'assert'
import solution from './CountEqualWithDividedNumber.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const nums = [4, 1, 2, 1, 4]
    const d = 2

    // when
    const result = solution(nums, d)

    // then
    assert.deepStrictEqual(1, result)
    done();
  });

  it('test 2', function (done) {
    // given
    const nums = [6, 5, 4, 3, 2, 1, 2, 3, 4, 5, 6]
    const d = 3

    // when
    const result = solution(nums, d)

    // then
    assert.deepStrictEqual(2, result)
    done();
  });

  it('test 3', function (done) {
    // given
    const nums = [2]
    const d = 2

    // when
    const result = solution(nums, d)

    // then
    assert.deepStrictEqual(0, result)
    done();
  });

  it('test 4', function (done) {
    // given
    const nums = []
    const d = 3

    // when
    const result = solution(nums, d)

    // then
    assert.deepStrictEqual(0, result)
    done();
  });

  it('test 5', function (done) {
    // given
    const nums = [10, 10]
    const d = 10

    // when
    const result = solution(nums, d)

    // then
    assert.deepStrictEqual(1, result)
    done();
  });

  it('test 6', function (done) {
    // given
    const nums = [10, 8, 6, 4, 2, 10, 8, 6, 4, 2]
    const d = 2

    // when
    const result = solution(nums, d)

    // then
    assert.deepStrictEqual(5, result)
    done();
  });
})
